ddtrace/appsec/remoteconfiguration.py

```diff
@@ -0,0 +1,58 @@
+import os
+
+from ddtrace.constants import APPSEC_ENV
+from ddtrace.constants import RC_ENV
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.remoteconfig import RemoteConfig
+from ddtrace.internal.remoteconfig.constants import ASM_FEATURES_PRODUCT
+from ddtrace.internal.utils.formats import asbool
+
+
+log = get_logger(__name__)
+
+
+def _rc_features_is_enabled():
+    return asbool(os.environ.get(RC_ENV)) and (
+        APPSEC_ENV not in os.environ or asbool(os.environ.get(APPSEC_ENV)) is True
+    )
+
+
+def appsec_enable_rc(tracer):
+    if _rc_features_is_enabled():
+        if not hasattr(tracer, "_rc"):
+            tracer._rc = RemoteConfig()
```

Comment:
This already implements a singleton pattern, so registering the product should be enough